tasks/queue-granules/index.js

```diff
@@ -95,22 +98,31 @@ async function queueGranules(event, testMocks = {}) {
   const executionArns = await pMap(
     groupedAndBatchedGranules,
     async (granuleBatchIn) => {
+      if (!granuleBatchIn[0].collectionId) {
+        deconstructedId = [granuleBatchIn[0].dataType, granuleBatchIn[0].version];
+      } else if (!granuleBatchIn[0].dataType && !granuleBatchIn[0].version) {
+        deconstructedId = deconstructCollectionId(granuleBatchIn[0].collectionId);
+      }
+
       const collectionConfig = await collectionConfigStore.get(
-        granuleBatchIn[0].dataType,
-        granuleBatchIn[0].version
+        deconstructedId[0], deconstructedId[1]
       );
 
       const createdAt = Date.now();
       const granuleBatch = updateGranuleBatchCreatedAt(granuleBatchIn, createdAt);
       await pMap(
         granuleBatch,
         (queuedGranule) => {
-          const collectionId = constructCollectionId(
-            queuedGranule.dataType,
-            queuedGranule.version
-          );
-
           const granuleId = queuedGranule.granuleId;
+          if ((!queuedGranule.dataType || !queuedGranule.version) && !queuedGranule.collectionId) {
+            throw new Error('Invalid collection, please check task input to make sure collection information is provided');
+          }
+
+          const collectionId = queuedGranule.collectionId !== undefined ? queuedGranule.collectionId
+            : constructCollectionId(
+              queuedGranule.dataType,
+              queuedGranule.version
+            );
```

Comment:
Because every granule in the batch is in the same collection, you can move this outside of the pMap loop and you don't need to throw the error you because you check when you create the batches.

```suggestion
          const collectionId = constructCollectionId(
            dataType,
            version
          );
```